python/doc/examples/probabilistic_modeling/random_vectors/plot_ratio_of_uniforms_exploitation.py

```diff
@@ -0,0 +1,455 @@
+"""
+Use the Ratio of Uniforms algorithm to sample a distribution
+============================================================
+"""
+
+# %%
+# In this example, we illustrate in dimension :math:`\inputDim = 1` the ratio of uniforms algorithm  to sample a distribution which
+# probability density function is known up to a normalization factor.
+#
+# Consider a scalar distribution whose
+# probability density function :math:`p` is known up to a normalization factor:
+# let :math:`f: \Rset \rightarrow \Rset^+` be a function such that
+# :math:`p(x) = cf(x)` for all :math:`x \in \Rset` with :math:`c \in \Rset^+_*`.
+#
+# Let :math:`r \in \Rset^+_*`. Let  :math:`A_{f,r}` be the domain defined by:
+#
+# .. math::
+#
+#    A_{f,r} = \left\{ (u, v) \in \Rset^2 \, |\, 0 \leq u \leq f\left(\dfrac{v}{u^r}\right)^{\frac{1}{1+r}}\right \}
+#
+# The Lebesgue measure of :math:`A_{f,r}`  is equal to :math:`\dfrac{1}{c(1+r)}`.
+#
+# Let :math:`(U, V)` be a random variable uniformly distributed on :math:`A_{f,r}`.
+# Then, :math:`X = \dfrac{V}{U^r}` is a random variable on
+# :math:`\Rset` distributed according to :math:`p`.
+#
+# Under the condition that :math:`f(x)^{\frac{1}{1+r}}` and
+# :math:`x f(x)^{\frac{1}{1+r}}` are bounded, then the domain
+# :math:`A_{f,r}` is bounded and can be included in a rectangular bounding box :math:`\tilde{A}_{f,r}`
+# defined by:
+#
+# .. math::
+#
+#    \tilde{A}_{f,r} & = \left[0, \sup_{x} f(x)^{\frac{1}{1+r}} \right] \times
+#    \left[ \inf_{x} x f(x)^{\frac{1}{1+r}} ,  \sup_{x} x f(x)^{\frac{1}{1+r}}\right]\\
+#                    & = \left[0, u_{sup} \right] \times \left[v_{inf}, v_{sup}\right]
+#
+# Note that :math:`u_{sup} > 0`, :math:`v_{sup} \geq 0` and  :math:`v_{inf} \leq 0`.
+#
+# This allows one to sample uniformly the domain :math:`A_{f,r}` by rejection sampling inside
+# :math:`\tilde{A}_{f,r}`.
+#
+# Study of the :math:`(u,v)`-space.
+# ---------------------------------
+# The boundary of :math:`\tilde{A}_{f,r}` is defined by:
+#
+# .. math::
+#
+#     \partial \tilde{A}_{f,r} = \cF_1 \cup \cF_2 \cup \cF_3 \cup \cF_4
+#
+# where the boundaries :math:`\cF_i` are defined by:
+#
+# .. math::
+#
+#    \cF_1 & = \left \{ (u, v_{sup}), 0 \leq u \leq u_{sup} \right \} \\
+#    \cF_2 & = \left \{ (u, v_{inf}), 0 \leq u \leq u_{sup} \right \}\\
+#    \cF_3 & = \left \{ (u_{sup}, v), v_{inf} \leq v \leq v_{sup} \right \}\\
+#    \cF_4 & = \left \{ (0, v), v_{inf} \leq v \leq v_{sup} \right \}
+#
+# The boundary of :math:`A_{f,r}` is defined by:
+#
+# .. math::
+#
+#    \partial A_{f,r}  = \left\{ (u,v) = \left( f(x)^{\dfrac{1}{1+r}}, x f(x)^{\dfrac{r}{1+r}} \right),
```

Comment:
\frac{1}{1+r}  and \frac{r}{1+r}   to avoid the large exponant ?

Reply:
yes